runtime/gc_glue_java/MarkingDelegate.cpp

```diff
@@ -600,62 +601,7 @@ MM_MarkingDelegate::setupReferenceObjectScanner(MM_EnvironmentBase *env, omrobje
 {
-	uintptr_t startIndex = 0;
-	uintptr_t headerBytesToScan = 0;
-	uintptr_t workItem = (uintptr_t)env->_workStack.peek(env);
-	if (PACKET_ARRAY_SPLIT_TAG == (workItem & PACKET_ARRAY_SPLIT_TAG)) {
-		Assert_MM_true(SCAN_REASON_PACKET == reason);
-		env->_workStack.pop(env);
-		/* since we are putting extra tagged objects on the work stack we are responsible for ensuring that the object scanned
-		 * count is correct.  The MM_MarkingScheme::scanObject code will increment _objectScanned by 1 for EVERY object
-		 * popped off of the work stack if the scan reason is SCAN_REASON_PACKET. This code is only executed during regular
-		 * packet scanning.
-		 */
-		env->_markStats._objectsScanned -= 1;
-		/* only mark the class the first time we scan any array */
-		startIndex = workItem >> PACKET_ARRAY_SPLIT_SHIFT;
-	} else {
-		/* account for header size on first scan */
-		headerBytesToScan = _extensions->indexableObjectModel.getHeaderSize((J9IndexableObject *)objectPtr);
-	}
-
-	uintptr_t slotsToScan = 0;
-	uintptr_t maxSlotsToScan = OMR_MAX(*sizeToDo / sizeof(fomrobject_t), 1);
-	Assert_MM_true(maxSlotsToScan > 0);
-	uintptr_t sizeInElements = _extensions->indexableObjectModel.getSizeInElements((J9IndexableObject *)objectPtr);
-	if (sizeInElements > 0) {
-		Assert_MM_true(startIndex < sizeInElements);
-		slotsToScan = sizeInElements - startIndex;
-
-		/* pointer arrays are split into segments to improve parallelism. split amount is proportional to array size
-		 * and inverse proportional to active thread count.
-		 * additionally, the less busy we are, the smaller the split amount, while obeying specified minimum and maximum.
-		 */
-
-		uintptr_t arraySplitSize = slotsToScan / (_extensions->dispatcher->activeThreadCount() + 2 * _markingScheme->getWorkPackets()->getThreadWaitCount());
-		arraySplitSize = OMR_MAX(arraySplitSize, _extensions->markingArraySplitMinimumAmount);
-		arraySplitSize = OMR_MIN(arraySplitSize, _extensions->markingArraySplitMaximumAmount);
-
-		if ((slotsToScan > arraySplitSize) || (slotsToScan > maxSlotsToScan)) {
-			slotsToScan = OMR_MIN(arraySplitSize, maxSlotsToScan);
-
-			/* immediately make the next chunk available for another thread to start processing */
-			uintptr_t nextIndex = startIndex + slotsToScan;
-			Assert_MM_true(nextIndex < sizeInElements);
-			void *element1 = (void *)objectPtr;
-			void *element2 = (void *)((nextIndex << PACKET_ARRAY_SPLIT_SHIFT) | PACKET_ARRAY_SPLIT_TAG);
-			Assert_MM_true(nextIndex == (((uintptr_t)element2) >> PACKET_ARRAY_SPLIT_SHIFT));
-			env->_workStack.push(env, element1, element2);
-			env->_workStack.flushOutputPacket(env);
-			MM_MarkJavaStats *markJavaStats = &(env->getGCEnvironment()->_markJavaStats);
-			markJavaStats->splitArraysProcessed += 1;
```

Comment:
It's good that you are moving this to OMR, but you'll have to move splitArraysProcessed and splitArraysAmount  to OMR Mark Stats.